public/app/features/templating/constant/reducer.ts

```diff
@@ -0,0 +1,27 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { ConstantVariableModel, VariableHide, VariableOption } from '../variable';
+import { EMPTY_UUID, getInstanceState } from '../state/types';
+import { createConstantOptionsFromQuery } from './actions';
+import { initialVariablesState } from '../state/variablesReducer';
+
+export const initialConstantVariableModelState: ConstantVariableModel = {
+  uuid: EMPTY_UUID,
+  global: false,
+  type: 'constant',
+  name: '',
+  hide: VariableHide.hideVariable,
+  label: '',
+  query: '',
+  current: {} as VariableOption,
+  options: [],
+  skipUrlSync: false,
+  index: -1,
+  initLock: null,
+};
+
+export const constantVariableReducer = createReducer(initialVariablesState, builder =>
+  builder.addCase(createConstantOptionsFromQuery, (state, action) => {
+    const instanceState = getInstanceState<ConstantVariableModel>(state, action.payload.uuid);
+    instanceState.options = [{ text: instanceState.query.trim(), value: instanceState.query.trim(), selected: false }];
+  })
+);
```

Comment:
is there a reason to createReducer vs createSlice here, fear of mutated state?

Reply:
Good question, no fear of mutation as both will use ImmerJs. We should use createSlice, good catch!

Reply:
Fixed, all reducers except the templatingReducer are now slices